nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection.js

```diff
@@ -1467,6 +1478,33 @@
                         }
                     });
 
+                    // determine whether or not to show the penalized icon
+                    connectionLabelContainer.select('text.penalized-icon')
+                        .classed('hidden', function () {
+                            if (d.permissions.canRead) {
+                                return d.component.flowFileAvailability !== 'HEAD_OF_QUEUE_PENALIZED';
```

Comment:
Looks like `flowFileAvailability` is part of the Connection status and not part of the Connection configuration. This makes sense as this value will change as the data is flowing. The path to `fileFileAvailability` should be `d.status.aggregateSnapshot.flowFileAvailability`. The convention is that the rendering status should be in `updateConnectionStatus` instead of `updateConnection`.